mlflow/tracking/_tracking_service/client.py

```diff
@@ -321,10 +321,12 @@ def download_trace_data(trace_info: TraceInfo) -> Optional[Trace]:
             try:
                 trace_data = self._download_trace_data(trace_info)
             except MlflowTraceDataException as e:
-                _logger.debug(
-                    f"Failed to download trace data for trace with {e.ctx}",
-                    trace_info.request_id,
-                    exc_info=True,
+                _logger.warning(
+                    (
+                        f"Failed to download trace data for trace {trace_info.request_id!r} "
+                        f"with {e.ctx}. For full traceback, set logging level to DEBUG."
+                    ),
+                    exc_info=_logger.isEnabledFor(logging.DEBUG),
```

Comment:
Changed `debug` to `warning` to avoid silently excluding traces and surprise users.